### ISSUE TITLE
After autoplay fail, no events fire

### ISSUE THREAD

**Issue by drewbaker** [vimeo/player.js#719]
### Expected Behavior

If auto play fails (due to no user input on initial load) then all events should actually work

### Actual Behavior

The browser notifies that `Autoplay is only allowed when approved by the user, the site is activated by the user, or media is muted.` and then no events from Vimeo API will ever fire.

Methods still work.

Works fine if you browse to the page without a reload and autoplay is successful. 

### Steps to Reproduce

Hard to reproduce this in codepen as it requires a static generated page. But I've tested this with `v2.15.0` and `2.14.1`.

```
           // Init player
           this.player = new VimeoPlayer(iframeEl, {autoplay: true})

           // Events, none of this work if autoplay fails.
           this.player.on("play", this.onPlay)
           this.player.on("playing", this.onPlay)
           this.player.on("pause", this.onPause)
           this.player.on("ended", this.onEnded)
```

**Comment by drewbaker** [comment]
OK, so if I add the events on a setTimeout to give time for the autoplay to fail, then the events work fine. So it seems the problem is adding events before an autoplay fail happens. Will continue to try workarounds...

**Comment by drewbaker** [comment]
OK, tried this after init of Player:

```
            if (this.autoplay) {
                // Try to auto play so we know if we have error
                try {
                    this.player.play()
                } catch (e) {
                    console.log("auto play failed ")
                    this.initEvents()
                }
            }
```            

But the catch never happens, so it's even more frustrating. Will try with the played() method, but not sure on timing of that.

**Comment by drewbaker** [comment]
OK, so the plot thickens... This will still not work with events. 
```
if(document.referrer) {
     this.player = new VimeoPlayer(iframeEl, {autoplay: true})
} else {
     this.player = new VimeoPlayer(iframeEl, {autoplay: false})
}

this.initEvents()
```

So now I suspect the issue is that I'm added the events too early? But docs says you don't need to wait till the player is ready to add events, but it sure seems like the only way to get this to work is to setTimeout() and wait an unknown amount of time before adding events...

**Comment by drewbaker** [comment]
This works:
```
if(document.referrer) {
     this.player = new VimeoPlayer(iframeEl, {autoplay: true})
} else {
     this.player = new VimeoPlayer(iframeEl, {autoplay: false})

    setTimeout(()=>{
        this.initEvents()
    }, 1000)
}
```

But it feels super hacky.

**Comment by jribal** [comment]
Hi, I also found a loading problem solved by waiting for `player.ready()` then wait 1 second.

Not really the same problem but it could be related.

I was trying to replace video inside the player multiple times. I settled on just calling `player.loadVideo()` but `player.enableTextTrack()` was failing with this error:
```
    Uncaught TypeError: Cannot read property 'label' of null
    at Object.<anonymous> (player.js:22)
    at Object.r (player.js:22)
    at Object.t.fire (player.js:22)
    at Object.<anonymous> (player.js:22)
    at t.fire (player.js:22)
    at Object.<anonymous> (player.js:22)
    at Object.t.fire (player.js:22)
    at player.js:22
```
If it tried to` player.unload() `or change the src of the video I could play another video but I got no more `'ended'` event
Waiting solved my problem of subtitles and event at once.

Thanks @drewbaker, I could not have thought of simply waiting.
